src/transformers/sylvester/main.py

```diff
@@ -0,0 +1,175 @@
+                    self._on_lend(decoded_params, tx_hash)
+                elif event["decoded"]["name"] == "Rent":
+                    self._on_rent(decoded_params, tx_hash)
+                elif event["decoded"]["name"] == "StopLend":
+                    self._on_stop_lend(decoded_params, tx_hash)
+                elif event["decoded"]["name"] == "StopRent":
+                    self._on_stop_rent(decoded_params, tx_hash)
+                elif event["decoded"]["name"] == "RentClaimed":
+                    self._on_rent_claimed(decoded_params, tx_hash)
+                    
+            logging.info(event)
+
+        self._flush_state = True
+
+    # todo: should be part of the interface
+    # todo: acts as the means to sync with db state
+    def update_memory_state(self) -> None:
+        """_summary_"""
+
+        if len(self._transformed) > 0:
+            return
+
+        state = self._db.get_all_items(self._db_name, self._collection_name)
+
+        if state is None:
+            return
+
+        self._transformed = state
+
+    # todo: should be part of the interface
+    def flush(self) -> None:
+        """_summary_"""
+
+        if self._flush_state:
+            # * write to the db
+            self._db.put_items(self._transformed, self._db_name, self._collection_name)
+            self._flush_state = False
+
+    def _on_rent_claimed(self, decoded_params, tx_hash: str) -> None:
+        # RentClaimed(uint256 indexed rentingID, uint32 collectedAt)
+
+        event = Event('RentClaimed', tx_hash, _RENT_CLAIMED, decoded_params)
+        self._transformed.append(event)
+
+    def _on_stop_rent(self, decoded_params, tx_hash: str) -> None:
+        # StopRent(indexed uint256 rentingID, uint32 stoppedAt)
+
+        event = Event('StopRent', tx_hash, _STOP_RENT, decoded_params)
+        self._transformed.append(event)
+
+    def _on_stop_lend(self, decoded_params, tx_hash: str) -> None:
+        # StopLend(uint256 indexed lendingID, uint32 stoppedAt)
+
+        event = Event('StopLend', tx_hash, _STOP_LEND, decoded_params)
+        self._transformed.append(event)
+
+    def _on_rent(self, decoded_params, tx_hash: str) -> None:
+        # Rent(indexed address renterAddress, indexed uint256 lendingID, indexed uint256 rentingID, 
+        # uint16 rentAmount, uint8 rentDuration, uint32 rentedAt)
+
+        event = Event('Rent', tx_hash, _RENT_FIELDS, decoded_params)
+        self._transformed.append(event)
+
+    def _on_lend(self, decoded_params, tx_hash: str) -> None:
+        # Lend(bool is721, indexed address lenderAddress, indexed address nftAddress, indexed uint256 tokenID, 
+        # uint256 lendingID, uint8 maxRentDuration, bytes4 dailyRentPrice, uint16 lendAmount, uint8 paymentToken)
+
+        event = Event('Lend', tx_hash, _LEND_FIELDS, decoded_params)
+        event['dailyRentPrice'] = unpack_price(event['dailyRentPrice'])
+
+        self._transformed.append(event)
+
+
+
+
+def hex_to_int(s):
```

Comment:
repeating same code, not good. We should move this out, and reuse from a single place.

Reply:
If we do that we will lose the package modularity since sylvester and azrael will need to share files. 

we could have a `utils.py` inside `src>transformers>shared` .. ? 

I would prefer to have duplicate code but preserve the 100% modularity we currently have. Let me know which you want.